src/operator/nn/pooling.cc

```diff
@@ -290,7 +290,7 @@ void PoolingComputeExCPU(const nnvm::NodeAttrs& attrs,
       workspace = &outputs[1];
     }
     DNNL_OPCHECK_INIT(false, 1, inputs, outputs);
-    DNNLPoolingCompute(ctx, param, inputs[0], req[0], outputs[0], workspace);
+    DNNLPoolingCompute(ctx, param, inputs[0], req[0], outputs[0], workspace, false);
```

Comment:
I wonder if we can run it like all others ops.
DNNL_OPCHECK_INIT...
DNNLRun(DNNLPoolingCompute...
DNNL_OPCHECK_RUN...

This one additional bool parameter can be passed by template argument and IsView check in checking support function won't be necessary. What do you think? + @bartekkuncer

Reply:
Making it consistent with other operators is desirable. @piotrwolinski-intel Please check if it introduces any problems.

Template here might make the code more readable as well, especially with `true/false` boolean parameter changed to simple `enum class PoolingType { regular, adaptive };`

Reply:
In my opinion as @piotrwolinski-intel is working currently on backward implementation for this op, this PR could be merged as is and the changes we are talking about can be applied along with backward propagation addition. Do you agree @bgawrych ?

Reply:
Ok, I'm merging it now and looking forward for backward refactor